src/components/typography/text/text.tsx

```diff
@@ -113,24 +125,25 @@ const Body = props =>
     </p>
   );
 Body.displayName = 'TextBody';
-Body.propTypes = {
-  isBold: PropTypes.bool,
-  isItalic: PropTypes.bool,
-  isInline: PropTypes.bool,
-  tone: PropTypes.oneOf([
-    'primary',
-    'secondary',
-    'information',
-    'positive',
-    'negative',
-    'inverted',
-  ]),
-  children: PropTypes.node.isRequired,
-  title: nonEmptyString,
-  truncate: PropTypes.bool,
+
+type ToneOfDetail =
+  | 'primary'
+  | 'secondary'
+  | 'information'
+  | 'positive'
+  | 'negative'
+  | 'inverted';
+type DetailProps = {
+  isBold?: boolean;
+  isItalic?: boolean;
+  isInline?: boolean;
+  tone?: ToneOfDetail;
+  title?: string;
+  truncate?: boolean;
+  children: React.ReactNode;
```

Comment:
We need to type the text components. Otherwise TypeScript will infer the types from prop-types (which it does) falsely and break compilation.